src/xds-proto/gen_build_yaml.py

```diff
@@ -0,0 +1,116 @@
+    proto_prefix: str
+    # Following 3 fields should be filled by build metadata from Bazel.
+    url: str = ''
+    hash: str = ''
+    strip_prefix: str = ''
+
+
+EXTERNAL_PROTO_LIBRARIES = {
+    'envoy_api':
+        ExternalProtoLibrary(destination='third_party/envoy-api',
+                             proto_prefix='third_party/envoy-api/'),
+    'com_google_googleapis':
+        ExternalProtoLibrary(destination='third_party/googleapis',
+                             proto_prefix='third_party/googleapis/'),
+    'com_github_cncf_udpa':
+        ExternalProtoLibrary(destination='third_party/udpa',
+                             proto_prefix='third_party/udpa/'),
+    'com_envoyproxy_protoc_gen_validate':
+        ExternalProtoLibrary(destination='third_party/protoc-gen-validate',
+                             proto_prefix='third_party/protoc-gen-validate/'),
+    'opencensus_proto':
+        ExternalProtoLibrary(destination='third_party/opencensus-proto/src',
+                             proto_prefix='third_party/opencensus-proto/src/'),
+}
+
+
+@dataclass()
+class _HttpArchive:
+    name: str = ''
+    url: str = ''
+    hash: str = ''
+    strip_prefix: str = ''
+
+
+def _fetch_raw_http_archives() -> ET.Element:
+    """Get xml output of bazel query invocation, parsed as XML tree"""
+    output = subprocess.check_output([
+        'tools/bazel', 'query', '--output', 'xml',
+        'kind(http_archive, //external:*)'
+    ])
+    return ET.fromstring(output)
+
+
+def _parse_http_archives(xml_tree: ET.Element) -> List[ExternalProtoLibrary]:
+    result = []
+    for xml_http_archive in xml_tree:
+        if xml_http_archive.tag != 'rule' or xml_http_archive.attrib[
+                'class'] != 'http_archive':
+            continue
+        http_archive = _HttpArchive()
+        for xml_node in xml_http_archive:
+            if xml_node.attrib['name'] == 'name':
+                http_archive.name = xml_node.attrib['value']
+            if xml_node.attrib['name'] == 'urls':
+                http_archive.url = xml_node[0].attrib['value']
+            if xml_node.attrib['name'] == 'url':
+                http_archive.url = xml_node.attrib['value']
+            if xml_node.attrib['name'] == 'sha256':
+                http_archive.hash = xml_node.attrib['value']
+            if xml_node.attrib['name'] == 'strip_prefix':
+                http_archive.strip_prefix = xml_node.attrib['value']
+        if http_archive.name not in EXTERNAL_PROTO_LIBRARIES:
+            continue
+        lib = EXTERNAL_PROTO_LIBRARIES[http_archive.name]
+        lib.url = http_archive.url
+        lib.hash = http_archive.hash
+        lib.strip_prefix = http_archive.strip_prefix
+        result.append(lib)
+    return result
+
+
+def main():
+    xml_tree = _fetch_raw_http_archives()
+    libraries = _parse_http_archives(xml_tree)
+    libraries.sort(key=lambda x: x.destination)
+    print(yaml.dump(dict(external_libraries=list(map(asdict, libraries)))))
```

Comment:
also, you're ambitiously calling this the new entity `external_library` but it's unclear what is the future plan with it.
Is it intended to be only used for importing .proto files (in which case it should be e.g. `external_proto_library`) or are you somehow planning to use it for other libraries in the future as well? If so, since there's no detail no plan on that, it's unclear to me whether that would actually work or not and if it would be useful (more research would be needed).